### DISCUSSION TITLE
Strange error - select \"ip_access\" from \"directus_roles\" where \"id\" is null limit $1

### DISCUSSION THREAD

**Discussion by ntatko** [directus/directus#4923] [Archive]
Hello!

I'm fairly new to directus, and I'm wondering what I may be doing wrong.

I'm calling a few endpoints within my directus api, some custom, others not, such as `POST` - `/auth/login`, `POST` - `/auth/refresh`, `GET` - `/items/courses`, `PATCH` - `/items/courses`, `GET` - `/custom/courses/location/0/0`. As you can see, no real pattern among them, and yet I'm getting consistent 500s back - 2 at a time, generally when the api hasn't been hit in over 5 minutes or so. I can reproduce this locally, it happens when running on a virtual machine.

The response body, following this error:
```json
{"errors":[{"message":"select \"ip_access\" from \"directus_roles\" where \"id\" is null limit $1 - Connection terminated unexpectedly","extensions":{"code":"INTERNAL_SERVER_ERROR"}}]}
```

And the server has a single line console response, green and ✨ as though nothing has gone wrong:
```bash
23:28:58 ✨ request errored POST 500 /custom/courses 577ms
```
☝️ I like this example, because I have logging around all errors and successes if it hits my custom logic at all, which it does not.

What might my issue be here?

**Comment by rijkvanzanten** [comment]
> Connection terminated unexpectedly

Sounds to me like the database goes to sleep / drops the connection after a certain time of inactivity 🤔

**Reply by WoLfulus** [reply]
This is exactly what I mentioned was happening to me a while ago @rijkvanzanten. @ntatko can you confirm if you're using PG? at least for me it happens with postgresql - haven't tested on other database vendors.

perhaps there's a way to set the connection to revalidate itself on knex or driver settings? should we send a ping query to the database every once in a while? doesn't sound like a proper "solution" for me though.

**Reply by ntatko** [reply]
Yep, PG (with postgis installed) is our db. We're using a docker image to run it - https://hub.docker.com/r/kartoza/postgis/

**Reply by rijkvanzanten** [reply]
A potential solution could be to set the pool size of directus to `min: 0` 🤔 That way, no connection will be kept open, which in turn could mean that that open connection isn't closed unexpectedly by the database

**Reply by ntatko** [reply]
Just to clarify, though, I doubt this is an issue with postgres, as we have 2 other APIs that get data from the API (we're migrating those to directus, props for making it so easy to have directus work so well with existing tables), but we've not seen this issue before, either using [Strapi](https://strapi.io/) or our own express server using [pg](https://www.npmjs.com/package/pg)

**Reply by rijkvanzanten** [reply]
Directus isn't much different from _express server using pg_ honestly. The only difference is that we rely on Knex which has a default pool size, hence my suggestion to try disabling that 🙂

**Reply by ntatko** [reply]
>A potential solution could be to set the pool size of directus to `min: 0` 🤔 

Would custom endpoints still be able to leverage the knex instance to get data?

**Reply by rijkvanzanten** [reply]
Sure, the difference is just how many connections knex keeps lingering for future use

**Reply by ntatko** [reply]
Where does one set `min: 0`? Is that in the `.env`?

**Reply by rijkvanzanten** [reply]
Yeah, using `DB_POOL_MIN` 👍🏻

**Reply by ntatko** [reply]
Hmm. Errors seem to be fewer. Might be worth adding a bug story somewhere to investigate this. I'll keep you informed.
